nixos/modules/services/web-servers/apache-httpd/default.nix

```
@@ -36,10 +36,13 @@ let
   dependentCertNames = unique (map (hostOpts: hostOpts.certName) acmeEnabledVhosts);
 
   mkListenInfo = hostOpts:
+    let
+      addrs = if hostOpts.listenAddresses != [] then hostOpts.listenAddresses else [ "*" ];
+    in
     if hostOpts.listen != [] then hostOpts.listen
-    else (
-      optional (hostOpts.onlySSL || hostOpts.addSSL || hostOpts.forceSSL) { ip = "*"; port = 443; ssl = true; } ++
-      optional (!hostOpts.onlySSL) { ip = "*"; port = 80; ssl = false; }
+    else flatten (
+      optional (hostOpts.onlySSL || hostOpts.addSSL || hostOpts.forceSSL) (map (addr: { ip = addr; port = 443; ssl = true; }) addrs) ++
+      optional (!hostOpts.onlySSL) (map (addr: { ip = addr; port = 80; ssl = false; }) addrs)
     );
```

```suggestion
    else
      optionals (hostOpts.onlySSL || hostOpts.addSSL || hostOpts.forceSSL) (map (addr: { ip = addr; port = 443; ssl = true; }) addrs) ++
      optionals (!hostOpts.onlySSL) (map (addr: { ip = addr; port = 80; ssl = false; }) addrs);
```

This should be equivalent.

---

I tested this out and without `flatten` a list of lists was produced... :thinking:

---

Even with plural `optional` -> `optionals`?

---

Oh... github should highlight that extra `s` :laughing: Let me look at that again.